common/src/main/java/tachyon/network/protocol/RPCMessageEncoder.java

```diff
@@ -42,9 +42,9 @@ protected void encode(ChannelHandlerContext ctx, RPCMessage in, List<Object> out
     long bodyBytes = 0;
     DataBuffer payload = null;
 
-    if (in.hasPayload()) {
-      payload = in.getPayloadDataBuffer();
-      bodyBytes = payload.getLength();
+    if (in instanceof PayloadMessage) {
+      payload = ((PayloadMessage)in).getPayloadDataBuffer();
+      bodyBytes = (payload == null) ? 0 : payload.getLength();
```

Comment:
Sure, you can close this for now, and maybe we can re-visit this at a later time if an interface would be beneficial.